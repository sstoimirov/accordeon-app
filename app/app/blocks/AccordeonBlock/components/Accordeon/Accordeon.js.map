{"version":3,"file":"Accordeon.js","sourceRoot":"","sources":["../../../../../src/blocks/AccordeonBlock/components/Accordeon/Accordeon.tsx"],"names":[],"mappings":";;;;;;AAAA,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAE/B,OAAO,EAAE,aAAa,EAAE,MAAM,kBAAkB,CAAC;AACjD,OAAO,EAAE,QAAQ,EAAE,MAAM,YAAY,CAAC;AAGtC,IAAa,SAAS,GAAtB,MAAa,SAAU,SAAQ,KAAK,CAAC,aAAgC;IAEjE,UAAU;QACN,OAAO,IAAI,CAAC,KAAK,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC,EAAE,EAAE,CAAC,EAAE,EAAE,CAAC,oBAAC,aAAa,IACjD,GAAG,EAAE,CAAC,EAAE,OAAO,EAAE,IAAI,CAAC,KAAK,CAAC,OAAO,CAAC,IAAI,CAAC,IAAI,EAAE,EAAE,CAAC,EAAE,CAAC,EACrD,EAAE,EAAE,EAAE,CAAC,EAAE,EACT,KAAK,EAAE,EAAE,MAAM,EAAE,EAAE,CAAC,KAAK,CAAC,MAAM,EAAE,MAAM,EAAE,EAAE,CAAC,KAAK,CAAC,MAAM,EAAE,EAC3D,QAAQ,EAAE,EAAE,CAAC,QAAQ,EACrB,OAAO,EAAE,EAAE,CAAC,OAAO,GAAI,CAC1B,CAAA;IACL,CAAC;IACD,MAAM;QACF,OAAO,CACH,6BAAK,SAAS,EAAC,iBAAiB;YAC5B,6BAAK,SAAS,EAAC,wBAAwB;gBACnC,6BAAK,SAAS,EAAC,2BAA2B,IACrC,IAAI,CAAC,UAAU,EAAE,CAChB,CACJ,CACJ,CACT,CAAA;IACL,CAAC;CACJ,CAAA;AAtBY,SAAS;IADrB,QAAQ;GACI,SAAS,CAsBrB;SAtBY,SAAS"}