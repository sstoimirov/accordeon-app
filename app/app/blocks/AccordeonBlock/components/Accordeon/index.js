export * from "./Accordeon";
//# sourceMappingURL=index.js.map