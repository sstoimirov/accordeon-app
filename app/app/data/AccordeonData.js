export const AccordeonData = [
    {
        id: "item_0",
        title: {
            opened: "Close",
            closed: "Open"
        },
        isOpened: false,
        content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        onClick: () => { }
    },
    {
        id: "item_1",
        title: {
            opened: "Close",
            closed: "Open"
        },
        isOpened: false,
        content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        onClick: () => { }
    },
    {
        id: "item_2",
        title: {
            opened: "Close",
            closed: "Open"
        },
        isOpened: false,
        content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        onClick: () => { }
    }
];
//# sourceMappingURL=AccordeonData.js.map