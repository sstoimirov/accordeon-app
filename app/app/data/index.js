export * from "./AccordeonData";
//# sourceMappingURL=index.js.map