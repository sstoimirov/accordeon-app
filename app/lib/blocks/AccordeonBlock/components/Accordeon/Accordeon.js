"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const AccordeonItem_1 = require("../AccordeonItem");
const mobx_react_1 = require("mobx-react");
let Accordeon = class Accordeon extends React.PureComponent {
    renderItem() {
        return this.props.items.map((el, i) => React.createElement(AccordeonItem_1.AccordeonItem, { key: i, onClick: this.props.onClick.bind(this, el.id), id: el.id, title: { opened: el.title.opened, closed: el.title.closed }, isOpened: el.isOpened, content: el.content }));
    }
    render() {
        return (React.createElement("div", { className: "accordeon-wrapp" },
            React.createElement("div", { className: "accordeon-wrapp--inner" },
                React.createElement("div", { className: "accordeon-elements-holder" }, this.renderItem()))));
    }
};
Accordeon = __decorate([
    mobx_react_1.observer
], Accordeon);
exports.Accordeon = Accordeon;
//# sourceMappingURL=Accordeon.js.map