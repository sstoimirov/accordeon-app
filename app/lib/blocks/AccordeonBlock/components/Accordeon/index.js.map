{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../../src/blocks/AccordeonBlock/components/Accordeon/index.ts"],"names":[],"mappings":";;;;;AAAA,iCAA4B"}