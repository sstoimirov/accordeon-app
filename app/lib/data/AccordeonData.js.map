{"version":3,"file":"AccordeonData.js","sourceRoot":"","sources":["../../src/data/AccordeonData.ts"],"names":[],"mappings":";;AAAa,QAAA,aAAa,GAAG;IACzB;QACI,EAAE,EAAE,QAAQ;QACZ,KAAK,EAAE;YACH,MAAM,EAAE,OAAO;YACf,MAAM,EAAE,MAAM;SACjB;QACD,QAAQ,EAAE,KAAK;QACf,OAAO,EAAE,yOAAyO;QAClP,OAAO,EAAE,GAAG,EAAE,GAAG,CAAC;KACrB;IACD;QACI,EAAE,EAAE,QAAQ;QACZ,KAAK,EAAE;YACH,MAAM,EAAE,OAAO;YACf,MAAM,EAAE,MAAM;SACjB;QACD,QAAQ,EAAE,KAAK;QACf,OAAO,EAAE,yOAAyO;QAClP,OAAO,EAAE,GAAG,EAAE,GAAG,CAAC;KACrB;IACD;QACI,EAAE,EAAE,QAAQ;QACZ,KAAK,EAAE;YACH,MAAM,EAAE,OAAO;YACf,MAAM,EAAE,MAAM;SACjB;QACD,QAAQ,EAAE,KAAK;QACf,OAAO,EAAE,yOAAyO;QAClP,OAAO,EAAE,GAAG,EAAE,GAAG,CAAC;KACrB;CACJ,CAAA"}