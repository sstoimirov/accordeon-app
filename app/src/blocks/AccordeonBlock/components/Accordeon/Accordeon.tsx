import * as React from "react";
import { AccordeonType } from "../../../../types";
import { AccordeonItem } from "../AccordeonItem";
import { observer } from "mobx-react";

@observer
export class Accordeon extends React.PureComponent<AccordeonType, {}>{

    renderItem() {
        return this.props.items.map((el, i) => <AccordeonItem
            key={i} onClick={this.props.onClick.bind(this, el.id)}
            id={el.id}
            title={{ opened: el.title.opened, closed: el.title.closed }}
            isOpened={el.isOpened}
            content={el.content} />
        )
    }
    render() {
        return (
            <div className="accordeon-wrapp">
                <div className="accordeon-wrapp--inner">
                    <div className="accordeon-elements-holder">
                        {this.renderItem()}
                    </div>
                </div>
            </div>
        )
    }
}
