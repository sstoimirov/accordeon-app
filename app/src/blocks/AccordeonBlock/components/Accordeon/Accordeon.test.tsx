import * as React from "react";
import { shallow, configure } from 'enzyme';
import { Accordeon } from "./Accordeon";
import { AccordeonData } from "../../../../data";
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });

describe("Accordeon", () => {

    it("Renders correclty with provided props", () => {
        const comp = shallow(<Accordeon items={AccordeonData} onClick={jest.fn()} />)
        expect(comp).toMatchSnapshot();
    })
})