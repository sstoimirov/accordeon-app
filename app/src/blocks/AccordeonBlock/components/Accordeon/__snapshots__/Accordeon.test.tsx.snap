// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordeon Renders correclty with provided props 1`] = `ShallowWrapper {}`;
